### PR TITLE
Bug 1563342 - google provider

### PR DESCRIPTION
**Pull request by djmitche**
This adds a google provider, including providing an instance token as instance identity proof.
    
This requires the latest (master) version of the go client; once the TC release after 14.2.0 is out, this can return to a released version.

### CODE REVIEW

File: provider/google/google.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		return err
+	}
+
+	workerIdentityProofMap := map[string]interface{}{"token": interface{}(proofToken)}
+	workerIdentityProof, err := json.Marshal(workerIdentityProofMap)
+	if err != nil {
+		return err
+	}
+
+	// We need an AWS Provisioner client for fetching taskcluster credentials.
```
Comment (imbstack) [inline provider/google/google.go]: This confuses me. Why do we need that?

Comment (djmitche) [reply provider/google/google.go]: womp, leftover comment.. will fix

File: provider/google/metadata_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			w.WriteHeader(200)
+			fmt.Fprintln(w, "42")
+		} else {
+			w.WriteHeader(404)
+			fmt.Fprintln(w, "Not Found")
+		}
+	}))
+	defer ts.Close()
+
+	metadataBaseURL = ts.URL + "/computeMetadata/v1/project"
```
Comment (milescrabill) [inline provider/google/metadata_test.go]: Why not create a new string here instead?

File: provider/google/metadata_test.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("Metadata-Flavor") != "Google" {
+			w.WriteHeader(400)
+			fmt.Fprintln(w, "Metadata-Flavor Missing")
+		} else if r.URL.Path == "/computeMetadata/v1/project/instance/attributes/taskcluster" {
+			w.WriteHeader(200)
+			fmt.Fprintln(w, `{"workerPoolId": "w/p"}`)
+		} else {
+			w.WriteHeader(404)
+			fmt.Fprintf(w, "Not Found: %s", r.URL.Path)
```
Comment (milescrabill) [inline provider/google/metadata_test.go]: Is the reason this is different between the two tests that you're emulating the real metadata service responses?

File: provider/google/metadata_test.go (hunk 2 above)
Comment (djmitche) [reply provider/google/metadata_test.go]: I'm not sure what you mean -- use `fmt.Sprintf`?

File: provider/google/metadata_test.go (hunk 3 above)
Comment (djmitche) [reply provider/google/metadata_test.go]: Yes.  I probably could use a single httptest.Server too, but this works.

File: provider/google/metadata_test.go (hunk 2 above)
Comment (milescrabill) [reply provider/google/metadata_test.go]: Never mind - this makes sense, I was misinterpreting what you were doing here. You need to use `metadataBaseURL`, it's what is used in the package.

**Pull request merged**